metrics/gauge.js

```diff
@@ -0,0 +1,8 @@
+var Gauge = module.exports = function Gauge(valFn) {
```

Comment:
Nice and straightforward,  I like it!   I would recommend renaming `this.valFn` to `value` in the case of the default gauge or just add a function prototype `value` that calls `valFn`.   For `CachedGauge` we could wrap the `value` function in something that only invokes `valFn` if enough time has passed, otherwise return a cached value of the last value returned from `valFn`